app/routes/goal_routes.py

```diff
@@ -1 +1,118 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, abort, make_response, request, Response
+from ..db import db
+from ..models.goal import Goal
+from ..models.task import Task
+from .route_utilities import validate_model
+from datetime import datetime, timezone
+
+goals_bp = Blueprint("goals_bp", __name__, url_prefix="/goals")
+
+
+@goals_bp.post("")
+def create_goal():
+
+    request_body = request.get_json()
+
+    try:
+        new_goal = Goal.from_dict(request_body)
+    except KeyError as e:
+        if e.args[0] == "title":
+            response = {"details": "Invalid data"}
+        return make_response(response, 400)
```

Comment:
It would be nice if we could provide more details about the invalid request body so that the client sending the request knows what about the body to fix.

```py
response = {"details": f"Invalid data: missing {e.args[0]}"}
```